components/script_plugins/unrooted_must_root.rs

```diff
@@ -142,26 +145,103 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for UnrootedPass {
 
         let def_id = cx.tcx.hir.local_def_id(id);
         let mir = cx.tcx.optimized_mir(def_id);
+        let mut unrooted_cx = UnrootedCx {
+            late_cx: cx,
+            def_id: def_id,
+            in_new_function: in_new_function,
+        };
```

Comment:
the main reason to define this was to define `is_unrooted_ty` as a method, so that it could access `self.def_id` to compute `generics_of`

Reply:
it may not be necessary, just an idea to keep code nicer and not pass 100 arguments

Reply:
I have implemented it as a method.